src/ansible_navigator/image_manager/puller.py

```diff
@@ -23,13 +24,15 @@ class ImagePuller:
     # pylint: disable=too-many-instance-attributes
     """Image puller"""
 
-    def __init__(self, container_engine: str, image: str, pull_policy: str):
+    def __init__(self, container_engine: str, image: str, arguments: List, policy: str):
         """Initialize the container image puller.
 
         :param container_engine: The name of the container engine
         :param image: The name of the image to pull
-        :param pull_policy: The current pull policy from the settings
+        :param arguments: Additional arguments to be appended to the pull policy
+        :param policy: The current pull policy from the settings
         """
+        self._arguments: List = arguments
```

Comment:
Can you give a more specific type constructor parameter here and above? Is it `List[str]`?

Reply:
yes, should be.